### PR TITLE
ci: add script to check git commits

### PR DESCRIPTION
**Pull request by rjarry**
Recently, there were multiple occasions where the git commits were not following the expected standards. Add a github action that will check every commit in all pull requests.

### CODE REVIEW

**Review by dsneddon — COMMENTED** [review]
I like where this is going, but apart from referencing this action script where can I find a formatting guide which specifies the conventions that should be followed? I don't think I've ever seen an OpenStack contributing guide that had rules quite as strict as those implemented here, so where did these conventions come from?

**Comment by rjarry** [comment]
The actual conventions are written up here:

https://git.sr.ht/~rjarry/aerc/tree/master/item/CONTRIBUTING.md
https://github.com/DPDK/grout/blob/main/CONTRIBUTING.md

**Review by karthiksundaravel — APPROVED** [review]
LGTM Thanks.

File: .github/workflows/check-commits [hunk 1]
```diff
@@ -0,0 +1,134 @@
+#!/bin/sh
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2024 Robin Jarry
+
```
Comment (vcandapp) [inline .github/workflows/check-commits:4]: @karthiksundaravel / @dsneddon - can we use copyright statement like this ? We have these in other files - eg; -https://github.com/os-net-config/os-net-config/blob/master/os_net_config/sriov_bind_config.py#L1

**Review by vcandapp — CHANGES_REQUESTED** [review]

File: .github/workflows/check-commits (hunk 1 above)
Comment (dsneddon) [reply .github/workflows/check-commits:4]: Perhaps @rjarry would be willing to make this available under the same license as the rest of os-net-config?

Comment (rjarry) [reply .github/workflows/check-commits]: This is the same license. Only a different copyright. I removed it. It makes no sense to add a license to such kind of script :)

**Review by karthiksundaravel — APPROVED** [review]
Happy with the changes. Thanks.

**Review by vcandapp — APPROVED** [review]
LGTM, thanks

**Pull request merged**